dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java

```diff
@@ -0,0 +1,182 @@
+/*
+ *    Copyright 2022 OICR and UCSC
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.client.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.dockstore.common.yaml.DockstoreYaml12;
+import io.dockstore.common.yaml.DockstoreYamlHelper;
+import io.dockstore.common.yaml.Service12;
+import io.dockstore.common.yaml.YamlWorkflow;
+import org.yaml.snakeyaml.Yaml;
+
+import static io.dockstore.client.cli.ArgumentUtility.out;
+
+/*
+    GENERAL STEPS:
+    1. Verfiy that YAML syntax is valid
+    2. Verify that all fields are named correctly
+    3. Verify that all required fields are present
+    4. Verify that all files exist
+ */
+
+public final class YamlVerifyUtility {
+
+    public static final String DOCKSTOREYML = ".dockstore.yml";
+    public static final String YAML = "yaml";
+
+    public static final String VALID_DOCKSTORE_YML = " is a valid dockstore yaml file and all required files are present";
+    public static final String INVALID_DOCKSTORE_YML = "Your .dockstore.yml is invalid:\n";
```

Comment:
Technically, all the `\n` usages should be `Platform.lineSeparator()`, but we don't officially support Windows, so I think you can ignore this, unless anybody else chimes in favor of.

Reply:
Yes, that's correct. Sorry, I don't know how/why I typed `Platform.lineSeparator()`